bridge/engine/EngineMixer.cpp

```diff
@@ -3834,15 +3838,57 @@ void EngineMixer::checkIfRateControlIsNeeded(const uint64_t timestamp)
     _probingVideoStreams = enableBEProbing;
 }
 
-void EngineMixer::runTransportTicks(const uint64_t timestamp)
+template <typename TStream>
+void removeIdleStreams(concurrency::MpmcHashmap32<size_t, TStream*>& streams,
+    std::function<void(TStream*)> removeMethod,
+    const uint64_t timestamp)
 {
-    for (auto videoIt : _engineVideoStreams)
+    for (const auto& it : streams)
     {
-        auto& videoStream = *videoIt.second;
-        videoStream.transport.runTick(timestamp);
+        auto stream = it.second;
+        if (!stream->idleTimeoutSeconds)
+        {
+            continue;
+        }
+
+        if (utils::Time::diffLT(stream->createdAt, timestamp, stream->idleTimeoutSeconds * utils::Time::sec))
+        {
+            continue;
+        }
+
+        const auto lastReceivedTimestamp = stream->transport.getLastReceivedPacketTimestamp();
+        if (utils::Time::diffGE(lastReceivedTimestamp, timestamp, stream->idleTimeoutSeconds * utils::Time::sec))
+        {
+            logger::info("Removing idle stream for endpoint %zu, last received packet timestamp: %zu, current: %zu",
+                "EngineMixer",
+                stream->endpointIdHash,
+                lastReceivedTimestamp,
+                timestamp);
+            removeMethod(stream);
+        }
+    }
+}
+
+#define REMOVE_IDLE_STREAMS(type)                                                                                      \
```

Comment:
Macro is something we should avoid as mush as we can on modern C++. I am sure we are able to similar without using templates

Reply:
Removed.